### PR TITLE
Add transaction view for solana and aptos.

### PR DESCRIPTION
**Pull request by moojing**
## Summary

- add transaction view for solana and aptos.
## Related Links

**Asana**:  https://app.asana.com/0/1201812548509877/1203488492251068/f

**Mockup**:  https://www.figma.com/file/X8Jx6GxcEB9gxoMxujdDSD/7.-Wallet-Web-2.0?node-id=129:18156&t=vFTbvBpfClSO6x5S-0

## Checklist

- [x] Pasted Asana or Mockup link.
- [x] Tagged labels.

## Screenshot/Gif

  
Screenshot 2023-02-02 at 3 53 28 PM
Screenshot 2023-02-02 at 3 53 31 PM
Screenshot 2023-02-02 at 5 04 21 PM

Screenshot 2023-02-02 at 5 06 20 PM

### CODE REVIEW

File: src/features/Aptos/Transaction/index.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  KEY_SESSION_ID,
+  KEY_USER_ID,
+  KEY_USER_TYPE,
+  getItem,
+} from "src/services/LocalStorage";
+
+import Connecting from "./components/Connecting";
+import Main from "./components/Main";
+
+// TODO: Add noncustodial component
```
Comment (yanglin5689446) [inline src/features/Aptos/Transaction/index.tsx:21]: it's ok to add todo notice, but please make them consistent across files, I suggest to make them all started with `@todo`

Comment (moojing) [reply src/features/Aptos/Transaction/index.tsx]: ok, I've change all the //todo comment to `@todo` [here](https://github.com/portto/wallet-web-ui/pull/30).

File: src/features/Aptos/Transaction/components/Main.tsx [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+const Main = () => {
+  const { context, send } = useTransactionMachine();
+  // @todo: add operation detection logic
+  const [recognizedTx] = useState(false);
+  // @todo: add operation verified logic
+  const [verifiedTx] = useState(false);
+
+  const { user, transaction, dapp } = context;
+  const dappDomain = new URL(dapp.url || "").host;
```
Comment (mordochi) [inline src/features/Aptos/Transaction/components/Main.tsx:26]: `""` in URL will cause an uncaught error

File: src/features/Aptos/Transaction/components/Main.tsx [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+      send({ type: "approve", data: { txHash: transactionHash } });
+      logSendTx({
+        domain: dappDomain,
+        url,
+        chain: blockchain,
+        type: "authz",
+        dAppName: name,
+        dAppId: id,
+      });
+    } else send({ type: "reject", data: { failReason: reason } });
```
Comment (mordochi) [inline src/features/Aptos/Transaction/components/Main.tsx:88]: There's no `failReason` in the defined transaction object. Do you mean `error`?

File: src/features/Aptos/Transaction/components/Main.tsx [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+              }
+              icon={
+                verifiedTx ? (
+                  <Check width="16px" height="16px" />
+                ) : (
+                  <CheckAlert width="16px" height="16px" />
+                )
+              }
+            >
+              {/* //TODO: add operation detection logic. */}
```
Comment (mordochi) [inline src/features/Aptos/Transaction/components/Main.tsx:210]: Missing to replace this `TODO` XD

File: src/features/Solana/Transaction/components/Main.tsx [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+      send({ type: "approve", data: { txHash: transactionHash } });
+      logSendTx({
+        domain: dappDomain,
+        url,
+        chain: blockchain,
+        type: "authz",
+        dAppName: name,
+        dAppId: id,
+      });
+    } else send({ type: "reject", data: { failReason: reason } });
```
Comment (mordochi) [inline src/features/Solana/Transaction/components/Main.tsx:77]: Same. Is this meant to be `error`?

**Review by mordochi — COMMENTED** [review]
I think this should verified tx? The UI might be this https://www.figma.com/file/X8Jx6GxcEB9gxoMxujdDSD/7.-Wallet-Web-2.0?node-id=135%3A18679&t=7fCeUz7d6rGTHn4L-1?
截圖 2023-02-06 下午1 39 08

File: src/features/Aptos/Transaction/components/Main.tsx (hunk 4 above)
Comment (moojing) [reply src/features/Aptos/Transaction/components/Main.tsx]: thanks for reminding https://github.com/portto/wallet-web-ui/pull/30/commits/056961631015cae195d3467541b9cda42c9050a0

File: src/features/Solana/Transaction/components/Main.tsx (hunk 5 above)
Comment (moojing) [reply src/features/Solana/Transaction/components/Main.tsx:77]: I'll handle this trouble in another [PR](https://github.com/portto/wallet-web-ui/commit/3286acd9f5a3f6539c4f6190ab8b417848da7838).

File: src/features/Aptos/Transaction/components/Main.tsx (hunk 2 above)
Comment (moojing) [reply src/features/Aptos/Transaction/components/Main.tsx]: fix [here](https://github.com/portto/wallet-web-ui/pull/30/commits/f258940914d0aedb34e9feb569c903f521a59f59) !

**Review by mordochi — APPROVED** [review]

**Pull request merged**